src/WebJobs.Script/Diagnostics/ILoggerFactoryBuilder.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Provides ways to plug into the ScriptHost ILoggerFactory initialization.
+    /// </summary>
+    public interface ILoggerFactoryBuilder
```

Comment:
Perhaps. For now, core SDK is different -- we let you build your own pipeline and give us the fully constructed LoggerFactory. We're more opinionated in the Functions world, but we could (and probably should) move that all down to the core SDK as well.